modules/common/vehicle_state/vehicle_state_provider.cc

```diff
@@ -234,7 +234,15 @@ math::Vec2d VehicleStateProvider::EstimateFuturePosition(const double t) const {
 math::Vec2d VehicleStateProvider::ComputeCOMPosition(
     const double rear_to_com_distance) const {
   // set length as distance between rear wheel and center of mass.
-  Eigen::Vector3d v(0.0, rear_to_com_distance, 0.0);
+  // TODO(Yu): the vehicle state transformation will be synconized with
+  // the planning reference transformation
+  Eigen::Vector3d v;
+  if (FLAGS_reverse_localization_transform &&
+      vehicle_state_.gear() == canbus::Chassis::GEAR_REVERSE) {
+    v << 0.0, 0.0, 0.0;
+  } else {
+    v << 0.0, rear_to_com_distance, 0.0;
+  }
```

Comment:
What you mean by planning reference transformation. If transform planning reference, there is no need to change com in vehicle state provider

Reply:
If currently we transform the planning reference to the COM, then all the control gains for forward driving have to be re-tuned. It may be conflict with all the current tests. 
With respect to your suggestion, I may consider to only transform the planning reference when backward driving (currently no one but us try the backward driving), to see if it produces the better effect.